hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/audit/EventAuditor.java

```diff
@@ -0,0 +1,124 @@
+        ignoringExceptions(() -> {
+            String createdObjectToString = objectMapper.writeValueAsString(createdObject);
+            AuditEvent event = new AuditEvent(AuditEventType.BEFORE_CREATION,
+                    createdObjectToString,
+                    createdObject.getClass().getSimpleName(),
+                    createdObject.toString(),
+                    username);
+            restTemplate.postForObject(eventDestination, event, Void.class);
+        });
+    }
+
+    @Override
+    public void beforeObjectRemoval(String username, String removedObjectType, String removedObjectName) {
+        ignoringExceptions(() -> {
+            AuditEvent event = new AuditEvent(AuditEventType.BEFORE_REMOVAL,
+                    removedObjectName,
+                    removedObjectType,
+                    removedObjectName,
+                    username);
+            restTemplate.postForObject(eventDestination, event, Void.class);
+        });
+    }
+
+    @Override
+    public void beforeObjectUpdate(String username, String objectClassName, Object objectName, PatchData patchData) {
+        ignoringExceptions(() -> {
+            String patchDataToString = objectMapper.writeValueAsString(patchData);
+            AuditEvent event = new AuditEvent(AuditEventType.BEFORE_UPDATE,
+                    patchDataToString,
+                    patchData.getClass().getSimpleName(),
+                    objectName.toString(),
+                    username);
+            restTemplate.postForObject(eventDestination, event, Void.class);
+        });
+    }
+
+    @Override
+    public void objectCreated(String username, Object createdObject) {
+        ignoringExceptions(() -> {
+            String createdObjectToString = objectMapper.writeValueAsString(createdObject);
+            AuditEvent event = new AuditEvent(AuditEventType.CREATED,
+                    createdObjectToString,
+                    createdObject.getClass().getSimpleName(),
+                    createdObject.toString(),
+                    username);
+            restTemplate.postForObject(eventDestination, event, Void.class);
+        });
+    }
+
+    @Override
+    public void objectRemoved(String username, String removedObjectType, String removedObjectName) {
+        ignoringExceptions(() -> {
+            AuditEvent event = new AuditEvent(AuditEventType.REMOVED,
+                    removedObjectName,
+                    removedObjectType,
+                    removedObjectName,
+                    username);
+            restTemplate.postForObject(eventDestination, event, Void.class);
+        });
+    }
+
+    @Override
+    public void objectUpdated(String username, Object oldObject, Object newObject) {
+        ignoringExceptions(() -> {
+            Diff diff = javers.compare(oldObject, newObject);
+            String diffToString = objectMapper.writeValueAsString(diff.toString());
```

Comment:
`org.javers.core.diff.Diff#toString` returns text like:
```
Diff:
* changes on pl.allegro.tech.hermes.api.Subscription/sample-subscription :
  - 'createdAt' changed from '2021-07-23T11:20:56.517Z' to ''
  - 'modifiedAt' changed from '2021-07-23T11:42:34.266Z' to ''
  - 'state' changed from 'ACTIVE' to 'SUSPENDED'
```

I know I mentioned earlier about serializing the payload to JSON, but I think there is no point of doing that in this case (actually, I am surprised that it works :)). So maybe here let's just use the result returned by `org.javers.core.diff.Diff#toString` as the payload. WDYT?

Reply:
You are right. That code was completely unnecessary and just `diff.toString()` is enough. I was also considering using something like `String diffToString = objectMapper.writeValueAsString(diff);`, but  in that case payload was too big, it contained all properties, not just changed ones, so it would be really hard to spot the difference.